web/middlewares/instance.go

```diff
@@ -70,3 +70,18 @@ func SetInstance() gin.HandlerFunc {
 		c.Set("instance", instance)
 	}
 }
+
+// GetInstance will return the instance linked to the given gin context
+func GetInstance(c *gin.Context) (*Instance, error) {
```

Comment:
In my own PR, I use c.MustGet("instance"), which panic if there is no instance. This avoids having to test if exists and returning an error in every handler.
  
If the instance middleware aborts when it cant make or get the instance for a request, it should be sufficiently robust. To be discussed